tests/main/services-after-before/task.yaml

```diff
@@ -20,3 +22,23 @@ execute: |
     echo "Service 'middle' is started before 'after-middle'"
     systemctl list-dependencies --plain --before snap.test-snapd-after-before-service.middle.service | \
         MATCH '\s*snap.test-snapd-after-before-service.after-middle.service'
+
+    START_LOG=/var/snap/test-snapd-after-before-service/common/start-log
+    function check_order() {
+        tr '\n' ',' < "$START_LOG" | \
+            MATCH "before-middle,middle,after-middle"
```

Comment:
Not quite sure the check is race free. Perhaps it would be better to look at `ExecMainStartTimestampMonotonic` property of individual service and compare?

Reply:
Hmm, yeah, this crossed my mind... I'll take a look at start timestamp, thanks.